tests/async_op_int/async_op_int.c

```diff
@@ -0,0 +1,142 @@
+    1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 4096, 8192
+};
+
+static const uint32_t context_sizes[] =
+{
+    /*powers of 2... */
+    1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 4096, 8192,
+
+    /*around 64/128/256 byte mutiple */
+    62, 63, 65, 66, 126,127, 129, 130, 254, 255, 257, 258,  
+
+    /*non power of 2*/
+    3, 5, 6, 7, 9, 10, 11, 12, 13, 14, 15, 17,
+
+    /*then some random numbers*/
+    345, 56, 487, 2829, 930, 881, 9230, 87309, 2173
+};
+
+/*this function exists only to have a real address somewhere, which is different from any other address*/
+static void ASYNC_OP_CANCEL_IMPL_1(void* context)
+{
+    (void)context;
+    /*doesn't do anything*/
+}
+
+static int ASYNC_OP_CANCEL_IMPL_2_ANTI_COMDAT = 0;
+/*this function exists only to have a real address somewhere, which is different from any other address*/
+static void ASYNC_OP_CANCEL_IMPL_2(void* context)
+{
+    ASYNC_OP_CANCEL_IMPL_2_ANTI_COMDAT += (context == NULL); /*just use context and ASYNC_OP_CANCEL_IMPL_2_ANTI_COMDAT somehow*/
+    /*doesn't do anything*/
+}
+
+static ASYNC_OP_CANCEL_IMPL cancels[] =
+{
+    NULL, 
+    ASYNC_OP_CANCEL_IMPL_1,
+    ASYNC_OP_CANCEL_IMPL_2
+};
+
+/*this function exists only to have a real address somewhere, which is different from NULL*/
+static void ASYNC_OP_DISPOSE_1(void* context)
+{
+    (void)context;
+    /*doesn't do anything*/
+}
+
+static int ASYNC_OP_DISPOSE_2_ANTI_COMDAT = 0;
+/*this function exists only to have a real address somewhere, which is different from any other address*/
+static void ASYNC_OP_DISPOSE_2(void* context)
+{
+    ASYNC_OP_DISPOSE_2_ANTI_COMDAT += (context == NULL); /*just use context and ASYNC_OP_DISPOSE_2_ANTI_COMDAT somehow*/
+}
+
+static ASYNC_OP_DISPOSE disposes[] =
+{
+    NULL,
+    ASYNC_OP_DISPOSE_1,
+    ASYNC_OP_DISPOSE_2
+};
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(setsBufferTempSize)
+{
+    ASSERT_ARE_EQUAL(int, 0, gballoc_hl_init(NULL, NULL));
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    gballoc_hl_deinit();
+}
+
+TEST_FUNCTION_INITIALIZE(function_init)
+{
+}
+
+TEST_FUNCTION_CLEANUP(cleans)
+{
+}
+
+TEST_FUNCTION(async_op_from_context_with_alignment_1)
+{
+    for (uint32_t i_align = 0; i_align < sizeof(context_aligns) / sizeof(context_aligns[0]); i_align++)
+    {
+        for (uint32_t i_size = 0; i_size < sizeof(context_sizes) / sizeof(context_sizes[0]); i_size++)
+        {
+            for (uint32_t i_cancel = 0; i_cancel < sizeof(cancels) / sizeof(cancels[0]); i_cancel++)
+            {
+                for (uint32_t i_dispose = 0; i_dispose < sizeof(disposes) / sizeof(disposes[0]); i_dispose++)
+                {
+                    ///arrange
+                    THANDLE(ASYNC_OP) async_op = async_op_create(cancels[i_cancel], context_sizes[i_size], context_aligns[i_align], disposes[i_dispose]);
+
+                    ASSERT_IS_NOT_NULL(async_op);
+                    void* context = async_op->context;

```

Comment:
>context [](http://example.com/codeflow?start=46&length=7)

should we maybe have something in the assert that validates the memory in context is valid? As in, we can read/write to it with a memcpy or something (which could likely AV if for example we didn't actually allocate enough memory).

Reply:
Good idea! Let me read it and write it :)

Reply:
I'll just do a memset because what is writeable is also readable in general for where we run our SW.